cmd/system-probe/config/adjust.go

```diff
@@ -47,6 +47,8 @@ func Adjust(cfg config.Config) {
 	adjustUSM(cfg)
 	adjustSecurity(cfg)
 
+	applyDefault(cfg, serviceDiscoveryNS("enabled"), true)
```

Comment:
I would suggest doing this also though, so the core agent has an accurate understanding if the module is supposed to be enabled or not.
```suggestion
    if config.IsContainerized() {
        cfg.Set(serviceDiscoveryNS("enabled"), false, model.SourceAgentRuntime)
    }
```